components/style/properties/longhand/ui.mako.rs

```diff
@@ -29,3 +29,38 @@ ${helpers.single_keyword("-moz-window-dragging", "default drag no-drag", product
                          gecko_enum_prefix="StyleWindowDragging",
                          animatable=False,
                          spec="None (Nonstandard Firefox-only property)")}
+
+<%helpers:longhand name="caret-color" animatable="False" products="none"
+                   spec="https://drafts.csswg.org/css-ui/#caret-color">
+    use std::fmt;
+    use style_traits::ToCss;
+    pub use self::computed_value::T as SpecifiedValue;
+
+    pub mod computed_value {
+        use cssparser::{Color, Parser};
+        use parser::{Parse, ParserContext};
+        use values::{Auto, Either};
+        use values::HasViewportPercentage;
+        use values::computed::ComputedValueAsSpecified;
+
+        no_viewport_percentage!(Color);
+        impl ComputedValueAsSpecified for Color {}
+
+        impl Parse for Color {
```

Comment:
We already have a `CSSColor` in `components/style/values/specified/` to aid with this :smile: